src/spikeinterface/widgets/unit_summary.py

```diff
@@ -107,15 +107,18 @@ def plot_matplotlib(self, data_plot, **backend_kwargs):
         # and use custum grid spec
         fig = self.figure
         nrows = 2
-        ncols = 3
+        ncols = 2
         if sorting_analyzer.has_extension("correlograms") or sorting_analyzer.has_extension("spike_amplitudes"):
             ncols += 1
+        if sorting_analyzer.has_extension("waveforms"):
```

Comment:
Sorry not related to this line but just above, two small things not from this diff but now seems a good time to fix:

- `            # UnitLocationsPlotter().do_plot(dp.plot_data_unit_locations, ax=ax1)` this commented line could be removed
- the variables `w` as output from both calls to `UnitLocationsWidget` is unused and could be removed.